device/network/network-Android.js

```diff
@@ -167,12 +168,35 @@ Network.createNotifier = function(params) {
     }
 };
 
-Network.__cancelAll = function() {
+Network.__cancelAll = function () {
     for (let i = 0; i < _instanceCollection.length; i++) {
         _instanceCollection[i].unsubscribe();
     }
 };
 
+Network.isConnected = function () {
+    return new Promise((resolve, reject) => {
+        const noConnection = Network.ConnectionType === Network.ConnectionType.NONE;
+        if (noConnection) {
+            return reject();
+        }
+        const http = new Http();
+        http.request({
+            url: "https://www.google.com",
```

Comment:
I would also check for parameter, Google might be blocked for some private networks

Reply:
Network.isConnected('turkcell.com/something')

Reply:
I added the parameter for this. Could you check it again? @ozcanovunc

Reply:
We might actually default to ifconfig.co or something that wouldn't be blocked by anything.